aspnetcore/signalr/scale.md

```diff
@@ -0,0 +1,87 @@
+
+## TCP connection resources
+
+The number of concurrent TCP connections that a web server can support is limited. Standard HTTP clients use *ephemeral* connections. These connections can be closed when the client goes idle and re-opened later. On the other hand, a SignalR connection is *persistent*. SignalR connections stay open even when the client goes idle. In a high-traffic app that serves many clients, these persistent connections can cause servers to hit their maximum number of connections.
+
+Persistent connections also consume some additional memory, to track each connection.
+
+The heavy use of connection-related resources by SignalR can affect other web apps that are hosted on the same server. When SignalR opens and holds the last available TCP connections, other web apps on the same server also have no more connections available to them.
+
+If a server runs out of connections, you'll see random socket errors and connection reset errors, for example:
+
+```
+An attempt was made to access a socket in a way forbidden by its access permissions...
+```
+
+To keep SignalR resource usage from causing errors in other web apps, run SignalR on different servers than your other web apps.
+
+To keep SignalR resource usage from causing errors in a SignalR app, scale out to limit the number of connections a server has to handle.
+
+## Scale out
+
+Unlike a stateless web app, you can't scale out a SignalR app simply by increasing the number of servers to handle increased traffic. New servers get new connections, and SignalR on server 2 knows nothing about the connections on servers 1 and 3. When SignalR on server 2 wants to send a message to all clients, the message only goes to the clients connected to that server. The following diagram illustrates this scenario.
+
+![Scaling SignalR without a backplane](scale/_static/scale-no-backplane.png)
+
+There are two recommended solutions to this problem:
+
+* Azure SignalR Service
+* Redis backplane
+
+
+## Azure SignalR Service
+
+The Azure SignalR Service is a proxy rather than a backplane. Each time a client initiates a connection to the server, the client is redirected to connect to the service. That process is illustrated in the following diagram:
+
+![Establishing a connection to the Azure SignalR Service](scale/_static/azure-signalr-service-one-connection.png)
+
+The result is that the service manages all of the client connections, while each server needs only a small constant number of connections to the service, as shown in the following diagram:
+
+![Clients connected to the service, servers connected to the service](scale/_static/azure-signalr-service-multiple-connections.png)
+
+This approach to scale-out has several advantages over the Redis backplane alternative:
+
+* The SignalR app can scale out without requiring sticky sessions, because clients are immediately redirected to the Azure service when they connect.
```

Comment:
The SignalR app should be `The SignalR service`. Since that's repeated mabe something like:

The SignalR service has several scale-out advantages over the Redis backplane. The SignalR service:

* can scale out without requiring sticky sessions, because clients are immediately redirected to the Azure service when they connect.
* can scale out based on the number of messages sent, while the Azure service automatically scales to handle any number of connections. For example, there could be thousands of clients, but if only a few messages per second are sent, the SignalR app won't need to scale out to multiple servers just to handle the connections themselves.
* won't use significantly more connection resources than a web app without SignalR.

Reply:
The bullets are about an app that uses SignalR, not about the SignalR service -- I'm rewriting to make that clearer.